### PR TITLE
[snowflake] implement snowflake connection interface

### PR DESCRIPTION
**Pull request by pavanka**
## Summary
- Implement the most basic interfaces necessary to run a simple malloy query on snowflake
- Write basic tests to check we can run an sql query and a malloy query via connection interface

## Test Plan
```
[nix-shell:~/code/malloy2]$ npx jest --runInBand packages/malloy-db-snowflake
 PASS  packages/malloy-db-snowflake/src/snowflake_connection.spec.ts
  db:Snowflake
    ✓ runs a SQL query (940 ms)
    ✓ runs a Malloy query (1808 ms)

 PASS  packages/malloy-db-snowflake/src/snowflake_executor.spec.ts
  db:SnowflakeExecutor
    ✓ verifies batch execute (793 ms)
    ✓ verifies stream iterable (352 ms)

Test Suites: 2 passed, 2 total
Tests:       4 passed, 4 total
Snapshots:   0 total
Time:        5.688 s
Ran all test suites matching /packages\/malloy-db-snowflake/i.
```

### CODE REVIEW

**Comment by pavanka** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1** <a href="https://app.graphite.dev/github/pr/datairisplatform/malloy2/1" target="_blank">Graphite</a> 
    * **PR #2** <a href="https://app.graphite.dev/github/pr/datairisplatform/malloy2/2" target="_blank">Graphite</a> 
      * **PR #3** <a href="https://app.graphite.dev/github/pr/datairisplatform/malloy2/3" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/datairisplatform/malloy2/3?utm_source=stack-comment).

File: packages/malloy-db-snowflake/package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "prepublishOnly": "npm run build"
   },
   "dependencies": {
-    "@malloydata/malloy": "^0.0.99",
+    "@malloydata/malloy": "file:../malloy",
```
Comment (amitaggarwal1) [inline packages/malloy-db-snowflake/package.json:24]: revert?

Comment (pavanka) [reply packages/malloy-db-snowflake/package.json:24]: yep this needs to be removed. though we need to think how we will deploy this to npm

**Comment by pavanka** [comment]
### Merge activity

* **Nov 9, 12:34 PM**: @pavanka started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/datairisplatform/malloy/3).

**Pull request merged**